odc/geo/cog/_rio.py

```diff
@@ -158,15 +159,27 @@ def _write_cog(
     def _write(pix, band, dst):
         if not use_windowed_writes:
             dst.write(pix, band)
-            return
-
-        for _, win in dst.block_windows():
-            if pix.ndim == 2:
-                block = pix[win.toslices()]
-            else:
-                block = pix[(slice(None),) + win.toslices()]
-
-            dst.write(block, indexes=band, window=win)
+        else:
+            for _, win in dst.block_windows():
+                if pix.ndim == 2:
+                    block = pix[win.toslices()]
+                else:
+                    block = pix[(slice(None),) + win.toslices()]
+
+                dst.write(block, indexes=band, window=win)
+
+        if tags is not None:
+            other_tags = {}
+            for tag, value in tags.items():
+                if tag == "scale":
+                    dst._set_all_scales([value])  # pylint: disable=protected-access
```

Comment:
There must be an official way to set those, no? Not a fan of this special handling of `scale,offset,units` tags. Those tags should probably be populated from attributes.

Reply:
From xarray attributes you mean?

I don't know why these are `_` protected access methods, but this is the way Vincent recommended to set them https://github.com/rasterio/rasterio/discussions/2933

Reply:
Looks like it should be `dst.{scales,offsets,units} = ...`

But, like I mentioned in the above, these should come from attributes, or maybe function parameters not from inside of "tags"

Reply:
Actually, turns out you can just assign the property like `dst.scales = [0.1]`.

Reply:
I put them in a `tags` dict to minimise the number of function parameters. If you think it makes sense for them to be handles separately to tags, I can do that. It'll be four parameters instead of one though.

Reply:
Also should not assume that value is not an array already, with the current implementation there is no way of specifying different values per band.

Another thing to confirm is what exactly happens when one sets `.scales|offsets`, is this just adding metadata  to the output (my assumption), or does this also turn on pixel value conversion.